src/DynamoCoreWpf/UI/HeaderTemplateSelector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using Dynamo.UI.Controls;
+
+namespace Dynamo.Controls
+{
+    public class HeaderTemplateSelector : DataTemplateSelector
```

Comment:
I would strongly encourage using the ```ListBox``` as it is when there is only one item in the list (instead of having a separate ```OneHeaderTemplate```) as described above. If there is a design limitation, can you please highlight it and let @elayabharath and myself know, so that we can make a meaningful *compromise* instead of making the code more complex?